src/go/wsl-helper/pkg/dnsmonitor/run.go

```diff
@@ -0,0 +1,121 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package dnsmonitor
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"net/netip"
+	"os"
+	"sort"
+
+	"golang.org/x/sys/windows"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+)
+
+func Run(path string) error {
+	addrs, err := winipcfg.GetAdaptersAddresses(windows.AF_UNSPEC, winipcfg.GAAFlagIncludeAll)
+	if err != nil {
+		return err
+	}
+	var activeInterfaces []*winipcfg.IPAdapterAddresses
+	for _, addr := range addrs {
+		if addr.OperStatus == winipcfg.IfOperStatusUp && addr.FirstDNSServerAddress != nil {
+			activeInterfaces = append(activeInterfaces, addr)
+		}
+	}
+
+	infMetric := func(a1, a2 *winipcfg.IPAdapterAddresses) bool {
+		return a1.Ipv4Metric < a2.Ipv4Metric
+	}
+	By(infMetric).Sort(activeInterfaces)
```

Comment:
```suggestion
	sort.Slice(activeInterfaces, func(i, j int) bool {
		return activeInterfaces[i].Ipv4Metric < activeInterfaces[j].Ipv4Metric
	})
```

Reply:
Also, we don't _actually_ need to sort; we just need the interface with the lowest metric. If we just do a dumb loop, that's O(n) instead of a full sort.  Not that we expect that many interfaces, but still.

Reply:
I just did the sort of the readability but sure, that works too.